dali/pipeline/pipeline.cc

```diff
@@ -334,16 +365,15 @@ int Pipeline::AddOperator(const OpSpec &const_spec, const std::string& inst_name
     std::string input_name = spec.InputName(input_idx);
     auto it = edge_names_.find(input_name);
 
-    DALI_ENFORCE(it != edge_names_.end(), "Input '" + input_name +
-        "' to op '" + spec.SchemaName() + "' is not known to the pipeline.");
-
-    string error_str = "(op: '" + spec.SchemaName() + "', input: '" +
-      input_name + "')";
+    DALI_ENFORCE(
+        it != edge_names_.end(),
+        make_string("Data node \"", input_name, "\" requested as ", FormatArgument(spec, arg_name),
+                    " to operator is not known to the pipeline."));
 
     if (!it->second.has_cpu) {
-      DALI_FAIL(make_string(
-          "Named arguments inputs to operators must be CPU data nodes. However, a GPU ",
-          "data node was provided (op: '", spec.SchemaName(), "', input: '", input_name, "')"));
+      DALI_FAIL(make_string("Error for ", FormatArgument(spec, arg_name),
+                            ". Named arguments inputs to operators must be CPU data nodes. "
+                            "However, a GPU data node was provided."));
```

Comment:
I don't like this message. It says that the input is GPU, but we're not really sure, are we? We know that it's not CPU. Perhaps there's some internal error and it's just empty? I think we should stick to the known facts and report those.
```suggestion
      DALI_FAIL(make_string("Error for ", FormatArgument(spec, arg_name),
                            ". Named arguments inputs to operators must be CPU data nodes. ""));
```
or
```suggestion
      assert(it->second.has_gpu);
      DALI_FAIL(make_string("Error for ", FormatArgument(spec, arg_name),
                            ". Named arguments inputs to operators must be CPU data nodes. "
                            "However, a GPU data node was provided."));
```

Reply:
Done